base_vat_optional_vies/models/res_partner.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2015 Tecnativa - Antonio Espinosa
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2019 FactorLibre - Rodrigo Bonilla
+# Copyright 2022 Moduon - Eduardo de Miguel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+
+from odoo.addons.base_vat.models.res_partner import _ref_vat
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    vies_passed = fields.Boolean(string="VIES validation", readonly=True)
+
+    @api.model
+    def simple_vat_check(self, country_code, vat_number):
+        res = super().simple_vat_check(country_code, vat_number)
+        partner = self._context.get("vat_partner")
+        if partner:
+            partner.update({"vies_passed": False})
+        return res
+
+    @api.model
+    def vies_vat_check(self, country_code, vat_number):
+        partner = self._context.get("vat_partner")
+        if partner:
+            # If there's an exception checking VIES, the upstream method will
+            # call simple_vat_check and thus the flag will be removed
+            partner.update({"vies_passed": True})
+        res = super().vies_vat_check(country_code, vat_number)
+        if res is False:
+            if partner:
+                partner.update({"vies_passed": False})
+            return self.simple_vat_check(country_code, vat_number)
+        elif partner:
+            partner.update({"vies_passed": True})
+        return res
+
+    @api.constrains("vat", "country_id")
+    def check_vat(self):
+        self.update({"vies_passed": False})
+        for partner in self:
+            partner = partner.with_context(vat_partner=partner)
+            super(ResPartner, partner).check_vat()
```

Comment:
Maybe can be improved to propagate vies_passed field to the children contacts:

```suggestion
		for partner in self.sorted(lambda p: bool(p.commercial_partner_id)):
			partner = partner.with_context(vat_partner=partner)
			if partner.commercial_partner_id:
				partner.update({"vies_passed": partner.commercial_partner_id.vies_passed})
			else:
				super(ResPartner, partner).check_vat()
```